tests/test_gui.py

```diff
@@ -0,0 +1,221 @@
+    }
+    new_project = shotgun.create("Project", project_data)
+
+    return new_project
+
+
+# This fixture will launch tk-run-app on first usage
+# and will remain valid until the test run ends.
+@pytest.fixture(scope="session")
+def host_application(sg_project):
+    """
+    Launch the host application for the Toolkit application.
+
+    TODO: This can probably be refactored, as it is not
+     likely to change between apps, except for the context.
+     One way to pass in a context would be to have the repo being
+     tested to define a fixture named context and this fixture
+     would consume it.
+    """
+    process = subprocess.Popen(
+        [
+            "python",
+            "-m",
+            "tk_toolchain.cmd_line_tools.tk_run_app",
+            # Allows the test for this application to be invoked from
+            # another repository, namely the tk-framework-widget repo,
+            # by specifying that the repo detection should start
+            # at the specified location.
+            "--location",
+            os.path.dirname(__file__),
+            "--context-entity-type",
+            sg_project["type"],
+            "--context-entity-id",
+            str(sg_project["id"]),
+        ]
+    )
+    try:
+        yield
+    finally:
+        # We're done. Grab all the output from the process
+        # and print it so that is there was an error
+        # we'll know about it.
+        stdout, stderr = process.communicate()
+        sys.stdout.write(stdout or "")
+        sys.stderr.write(stderr or "")
+        process.poll()
+        # If returncode is not set, then the process
+        # was hung and we need to kill it
+        if process.returncode is None:
+            process.kill()
+        else:
+            assert process.returncode == 0
+
+
+@pytest.fixture(scope="session")
+def app_dialog(host_application):
+    """
+    Retrieve the application dialog and return the AppDialogAppWrapper.
+    """
+    before = time.time()
+    while before + 60 > time.time():
+        if sgtk.util.is_windows():
+            app_dialog = AppDialogAppWrapper(topwindows)
+        else:
+            app_dialog = AppDialogAppWrapper(topwindows["python"])
+
+        if app_dialog.exists():
+            yield app_dialog
+            app_dialog.close()
+            return
+    else:
+        raise RuntimeError("Timeout waiting for the app dialog to launch.")
+
+
+class AppDialogAppWrapper(object):
+    """
+    Wrapper around the app dialog.
+    """
+
+    def __init__(self, parent):
+        """
+        :param root:
+        """
+        self.root = parent["Shotgun: Shotgun Python Console"].get()
+
+    def exists(self):
+        """
+        ``True`` if the widget was found, ``False`` otherwise.
+        """
+        return self.root.exists()
+
+    def close(self):
+        self.root.buttons["Close"].get().mouseClick()
+
+
+def test_python_console(app_dialog):
+    """
+    Quick UI validation of the Python Console
```

Comment:
It would be good if the description described the rough process or areas of the console it is going to check, to save having to read through all the code to find out.

Also I can't say I'm a fan of all the tests being covered in one method. This may not be possible, but can any of these be broken out into separate tests?